### PR TITLE
DRIVERS-2616 Fix path handling for Windows

### PR DESCRIPTION
**Pull request by blink1073**
Tested with https://github.com/mongodb/mongo-python-driver/pull/1365

### CODE REVIEW

File: .evergreen/auth_oidc/oidc_get_tokens.py [hunk 1]
```diff
@@ -18,26 +18,26 @@ def main():
...
+    config['token_file'] = join(token_dir, 'test_user2')
     get_id_token(config)
     for i in range(2):
-        config['token_file'] = os.path.join(token_dir, f'test_user2_{i+1}')
+        config['token_file'] = join(token_dir, f'test_user2_{i+1}')
         get_id_token(config)
     config['issuer'] = secrets['oidc_issuer_1_uri']
     config['username'] = 'test_user1'
-    config['token_file'] = os.path.join(token_dir, 'test_user1_expires')
+    config['token_file'] =  join(token_dir, 'test_user1_expires')
```
Comment (NoahStapp) [inline .evergreen/auth_oidc/oidc_get_tokens.py:36]: There's an extra space here.

File: .evergreen/auth_oidc/oidc_get_tokens.sh [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 fi
-export OIDC_TOKEN_DIR=${OIDC_TOKEN_DIR:-/tmp/tokens}
-
-rm -rf authoidcvenv
+mkdir -p $OIDC_TOKEN_DIR
 . ./activate-authoidcvenv.sh
-python oidc_get_tokens.py
\ No newline at end of file
+python oidc_get_tokens.py
+ls -ltr $OIDC_TOKEN_DIR
```
Comment (NoahStapp) [inline .evergreen/auth_oidc/oidc_get_tokens.sh:16]: What's the purpose of this `ls`?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/auth_oidc/oidc_get_tokens.sh (hunk 2 above)
Comment (blink1073) [reply .evergreen/auth_oidc/oidc_get_tokens.sh:16]: It was for debugging, I'll remove it.

File: .evergreen/auth_oidc/oidc_get_tokens.py (hunk 1 above)
Comment (blink1073) [reply .evergreen/auth_oidc/oidc_get_tokens.py]: Fixed

File: .evergreen/auth_aws/lib/aws_handle_oidc_creds.py [hunk 3]
```diff
@@ -105,6 +105,7 @@ def get_id_token(config=None, expires=None):
     token = response["id_token"]
     if config['token_file']:
         with open(config['token_file'], 'w') as fid:
+            print(f"Writing token file: {config['token_file']}")
```
Comment (NoahStapp) [inline .evergreen/auth_aws/lib/aws_handle_oidc_creds.py:108]: Is this intended to be used for debugging?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/auth_aws/lib/aws_handle_oidc_creds.py (hunk 3 above)
Comment (blink1073) [reply .evergreen/auth_aws/lib/aws_handle_oidc_creds.py:108]: Yes, it was too opaque before.

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**